dist/angular-date-picker.js

```diff
@@ -22,92 +22,121 @@ angular.module( "vokal.datePicker", [] )
     {
         "use strict";
 
-        var defaultFormat = "M/d/yyyy";
+        var defaultFormat = "M/D/YYYY";
 
-        function validateDate( date )
+        function validateDate( dateOrString )
         {
-            return !!date && angular.isFunction( date.getTime ) && !isNaN( date.getTime() );
-        }
-        function convertToDate( str )
-        {
-            return validateDate( str ) ? str : new Date( str );
+            return !!dateOrString && moment( new Date( dateOrString ) ).isValid();
         }
 
         return {
             restrict: "A",
-            scope: {},
+            scope: {
+                model: "=ngModel",
+                timezone: "="
+            },
             require: "ngModel",
             link: function ( scope, element, attrs, ngModelController )
             {
-                var localDate = new Date( new Date().toDateString() );
-                function filterOutput( date )
+                if( scope.timezone && !moment.tz )
+                {
+                    console.warn( "Trying to use moment-timezone without including the script." );
```

Comment:
I'd lean towards maintaining the optional timezone support as-is.  Certainly fixing it so the warning here warns without then immediately throwing an error, like you said you'd do, but then supporting timezones should you opt to include the 2010-2020 file (the full one is absurd).  I see the warning as just a reminder "hey you forgot this" or rather "hey don't send a timezone, you aren't using them remember."

It's not really a "dependency" as long as you write it to work error-free with or without it, it's just additional functionality you can choose to include should you want it.